src/libdredd/src/mutate_visitor.cc

```diff
@@ -278,6 +279,29 @@ bool MutateVisitor::VisitBinaryOperator(
   return true;
 }
 
+bool MutateVisitor::VisitExpr(clang::Expr* expr) {
+  if (!IsInFunction()) {
+    // Only consider mutating expressions that occur inside functions.
+    return true;
+  }
+
+  // For unary and binary operators, unary operator insertion and
+  // statement replacement are handled in these classes.
+  if (llvm::dyn_cast<clang::BinaryOperator>(expr) != nullptr ||
+      llvm::dyn_cast<clang::UnaryOperator>(expr) != nullptr) {
+    return true;
+  }
+
+  if (expr->isLValue() ||
```

Comment:
The decision to ignore l-values deserves a comment. Also, as per my thoughts from having used this a bit, I think unary operator insertion could still be valuable for non-boolean l-values in C++ (using the ++ and -- operators).

Reply:
I have changed this to follow your comments above.